UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Services/FssEventProcessorTest.cs

```diff
@@ -0,0 +1,165 @@
+        private FssEventProcessor _fssEventProcessor;
+        private CustomEventGridEvent _fakeCustomEventGridEvent;
+        private FssEventData _fakeFssEventData;
+        public const string correlationId = "7b838400-7d73-4a64-982b-f426bddc1296";
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakeFssEventValidationAndMappingService = A.Fake<IFssEventValidationAndMappingService>();
+            _fakeLogger = A.Fake<ILogger<FssEventProcessor>>();
+            _fakeAzureEventGridDomainService = A.Fake<IAzureEventGridDomainService>();
+            _fakeCustomEventGridEvent = GetCustomEventGridEvent();
+            _fakeFssEventData = GetFssEventData();
+
+            _fssEventProcessor = new FssEventProcessor(_fakeFssEventValidationAndMappingService,
+                                                       _fakeLogger, _fakeAzureEventGridDomainService);
+        }
+
+        [Test]
+        public void  WhenValidInRequest_ThenReceiveEventType()
+        {
+            string result = _fssEventProcessor.EventType;
+
+            Assert.AreEqual(_fakeCustomEventGridEvent.Type, result);
+        }
+
+        [Test]
+        public async Task WhenInvalidPayloadInRequest_ThenReceiveSuccessfulResponse()
+        {
+            var validationMessage = new ValidationFailure("BatchId", "BatchId cannot be blank or null.")
+            {
+                ErrorCode = HttpStatusCode.OK.ToString()
+            };
+
+            A.CallTo(() => _fakeFssEventValidationAndMappingService.ValidateFssEventData(A<FssEventData>.Ignored))
+                            .Returns(new ValidationResult(new List<ValidationFailure> { validationMessage }));
+
+            ExternalNotificationServiceProcessResponse result = await _fssEventProcessor.Process(_fakeCustomEventGridEvent, correlationId);
+
+            Assert.AreEqual("BatchId cannot be blank or null.", result.Errors.Single().Description);
+            Assert.AreEqual(HttpStatusCode.OK, result.StatusCode);
+        }
+
+        [Test]
+        public async Task WhenInvalidBusinessUnitInRequest_ThenReceiveSuccessfulResponse()
+        {
+            CancellationToken cancellationToken = CancellationToken.None;
+            CloudEvent cloudEvent = new("test", "test", new object());
+            _fakeFssEventData.BusinessUnit = "test";
+
+            A.CallTo(() => _fakeFssEventValidationAndMappingService.ValidateFssEventData(A<FssEventData>.Ignored)).Returns(new ValidationResult());
+
+            A.CallTo(() => _fakeFssEventValidationAndMappingService.FssEventDataMapping(A<CustomEventGridEvent>.Ignored, A<string>.Ignored)).Returns(cloudEvent);
+
+            A.CallTo(() => _fakeAzureEventGridDomainService.JsonDeserialize<FssEventData>(A<object>.Ignored)).Returns(_fakeFssEventData);
+
+            A.CallTo(() => _fakeAzureEventGridDomainService.PublishEventAsync(A<CloudEvent>.Ignored, A<string>.Ignored, cancellationToken));
```

Comment:
Use defined object instead of fake object.
```suggestion
            A.CallTo(() => _fakeAzureEventGridDomainService.PublishEventAsync(cloudEvent, correlationId, cancellationToken));
```

Reply:
Thanks for highlighting, done required changes.